src/main/scala/mesosphere/marathon/stream/CollectionStage.scala

```diff
@@ -0,0 +1,48 @@
+package mesosphere.marathon.stream
+
+import akka.stream.{ Attributes, Inlet, SinkShape }
+import akka.stream.stage.{ GraphStageLogic, GraphStageWithMaterializedValue, InHandler }
+
+import scala.collection.mutable
+import scala.concurrent.{ Future, Promise }
+
+/**
+  * Akka Streaming Graph Stage that collects a set of values into the given collection
+  * Based on akka's SeqStage
+  */
+private final class CollectionStage[T, C](buf: mutable.Builder[T, C])
```

Comment:
This was the "coolest" change and reminded me why the collection API is so nice. Wish akka did this instead of only doing it for seq.